android-kotlin/app/src/main/java/com/mongodb/app/LoginActivity.kt

```diff
@@ -0,0 +1,125 @@
+package com.mongodb.app
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import io.realm.mongodb.Credentials
+/*
+* LoginActivity: launched whenever a user isn't already logged in. Allows a user to enter email
+* and password credentials to log in to an existing account or create a new account.
+*/
+class LoginActivity : AppCompatActivity() {
+    private lateinit var username: EditText
+    private lateinit var password: EditText
+    private lateinit var loginButton: Button
+    private lateinit var accountExistsToggleButton: Button
+    private var createUser = true
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        username = findViewById(R.id.input_username)
+        password = findViewById(R.id.input_password)
+        loginButton = findViewById(R.id.button_create_login)
+        accountExistsToggleButton = findViewById(R.id.button_existing_account)
+
+        loginButton.setOnClickListener { login() }
+        accountExistsToggleButton.setOnClickListener { toggleExistingUser() }
+
+        loginButton.isEnabled = true
+        accountExistsToggleButton.isEnabled = true
+    }
+
+    override fun onBackPressed() {
+        // Disable going back to the MainActivity
+        moveTaskToBack(true)
+    }
+
+    private fun onLoginSuccess() {
+        // successful login ends this activity, bringing the user back to the project activity
+        startActivity(Intent(this, ItemsActivity::class.java))
+        // finish()
+    }
+
+    private fun onLoginFailed(errorMsg: String) {
+        Log.e(TAG(), errorMsg)
+        Toast.makeText(baseContext, errorMsg, Toast.LENGTH_LONG).show()
+    }
+
+    // Function to validate that the username adheres to all username rules.
+    private fun validateUsername(username: String): Boolean {
+        if (username.isEmpty()) {
```

Comment:
I don't know if we want to include username validation at all. The server will do some validation.

By the way, you might want to use `isBlank()` here.